### PR TITLE
Add (shallow) product SBOM example using CycloneDX

### PR DESCRIPTION
**Pull request by mprpic**
Also add a previously missing CPE for the Appstream variant of RHEL 9.2 EUS.

Resolves SBOMER-208.

### CODE REVIEW

**Comment by mprpic** [comment]
@jonathanchristison @twaugh @goldmann @jasinner Review/comments welcome!

File: sbom/examples/product/create_product_sbom.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    "content": checksum.split(":")[1],
+                }
+                for checksum in pkg.checksums
+            ],
+        }
+        sbom["components"].append(component)
+
+    for purl in pkg.purls:
+        dep = {"ref": purl, "dependsOn": [cpe for cpe in product.cpes]}
+        sbom["dependencies"].append(dep)
```
[the remark this answers is not in the document]
Comment (mprpic) [reply sbom/examples/product/create_product_sbom.py:174]: Do you know what the relationship looks like for either case in guac? Would it change the way we query for "show me pkg X in product Y"?

Maybe also a question for @jonathanchristison @mertbugrabicak.

File: sbom/examples/product/rhel-9.2-eus.cdx.json [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+          "content": "31b5079268339cff7ba65a0aee77930560c5adef4b1b3f8f5927a43ee46a56d9"
+        }
+      ]
+    }
+  ],
+  "dependencies": [
+    {
+      "ref": "pkg:rpm/redhat/openssl@3.0.7-18.el9_2?arch=src&repository_id=rhel-9-for-aarch64-baseos-eus-source-rpms",
+      "dependsOn": [
+        "cpe:/a:redhat:rhel_eus:9.2::appstream",
```
Comment (jasinner) [inline sbom/examples/product/rhel-9.2-eus.cdx.json:448]: I think we could improve this by only adding a relationship to the baseos variant for packages in the baseos repo.

File: sbom/examples/product/create_product_sbom.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasinner) [reply sbom/examples/product/create_product_sbom.py:174]: From what I can tell Guac is not able to handle nested components in CDX. The nested components didn't show up as Packges at all when uploading an SBOM with nested components [like this](https://gist.github.com/jasinner/f04d3b562d5c211df2fa5c46fabac96d).

They did show up as Packages when using dependencies as per this PR.

[the remark this answers is not in the document]
Comment (mertbugrabicak) [reply sbom/examples/product/create_product_sbom.py:174]: Nested components support for CycloneDX is a feature that's currently being worked on in our downstream of GUAC and should be available eventually as part of 1.2 (currently seems to have some bugs)

File: sbom/examples/product/rhel-9.2-eus.cdx.json (hunk 2 above)
Comment (mprpic) [reply sbom/examples/product/rhel-9.2-eus.cdx.json:448]: Note that this is the shallow product manifest, so it only provides the relationship of all product variants for a given product version in an advisory to the root component provided by that advisory (SRPM). In the RPM manifest (i.e. https://github.com/RedHatProductSecurity/security-data-guidelines/blob/main/sbom/examples/rpm/release/openssl-3.0.7-18.el9_2.spdx.json), we do have the repo IDs mapped to all specific binary RPMs and the root component as well. Though, now that I'm looking at the repo IDs, they're all baseos ones, so I should fix e.g.:

https://github.com/RedHatProductSecurity/security-data-guidelines/blob/main/sbom/examples/rpm/release/openssl-3.0.7-18.el9_2.spdx.json#L166

to be:

```
pkg:rpm/redhat/openssl-perl@3.0.7-18.el9_2?arch=aarch64&repository_id=rhel-9-for-aarch64-appstream-rpms
```

**Pull request merged**